osu.Game/Screens/Play/Player.cs

```diff
@@ -197,6 +205,19 @@ private void load(AudioManager audio, OsuConfigManager config)
                 skipOverlay.Hide();
             }
 
+            if (game != null)
+                OverlayActivationMode.BindTo(game.OverlayActivationMode);
+
+            gameplayOverlaysDisabled.ValueChanged += disabled =>
+            {
+                if (DrawableRuleset.HasReplayLoaded.Value)
+                    OverlayActivationMode.Value = OverlayActivation.UserTriggered;
+                else
+                    OverlayActivationMode.Value = disabled.NewValue && !DrawableRuleset.IsPaused.Value ? OverlayActivation.Disabled : OverlayActivation.UserTriggered;
+            };
+            DrawableRuleset.IsPaused.BindValueChanged(_ => gameplayOverlaysDisabled.TriggerChange());
+            DrawableRuleset.HasReplayLoaded.BindValueChanged(_ => gameplayOverlaysDisabled.TriggerChange());
```

Comment:
this is super awkwardly written to me; if you wanted to centralise logic of changing `OverlayActivationMode` I would suggest to follow the example of `updatePauseOnFocusLostState()` below by extracting the callback to a `updateOverlayActivationMode()` method, and then using it in the callbacks of all these three bindables - this way there's no weird daisy-chaining of value changed callbacks